lib/ansible/module_utils/basic.py

```diff
@@ -1413,14 +1413,17 @@ def _check_locale(self):
             self.fail_json(msg="An unknown error was encountered while attempting to validate the locale: %s" %
                            to_native(e), exception=traceback.format_exc())
 
-    def _handle_aliases(self, spec=None, param=None):
+    def _handle_aliases(self, spec=None, param=None, option_prefix=''):
         if spec is None:
             spec = self.argument_spec
         if param is None:
             param = self.params
 
         # this uses exceptions as it happens before we can safely call fail_json
-        alias_results, self._legal_inputs = handle_aliases(spec, param)
+        alias_warnings = []
+        alias_results, self._legal_inputs = handle_aliases(spec, param, alias_warnings=alias_warnings)
+        for option, alias in alias_warnings:
+            self._warnings.append('Both option %s and its alias %s are set.' % (option_prefix + option, option_prefix + alias))
```

Comment:
its internal, but i would still use the warn method instead of accessing the list directly

Reply:
The reason was that `self.warn()` calls `self.log()`, and `no_log` isn't initialized at this point. That is probably not a problem because we are not outputting values, but it might be better to avoid calling `self.warn()` before someone later adds the value here and suddenly no-log values are dumped into the logs.

Reply:
ah, missed that, good point, nvmd me.